### ISSUE TITLE
bug: 4.4.1 regression - defaultHref of ion-back-button is not working with query params

### ISSUE THREAD

**Issue by astec** [ionic-team/ionic#18354]
# Bug Report

**Ionic version:**

[x] **4.4.1**

**Current behavior:**

Back button stopped to work for deep-linked pages if `defaultHref` includes query parameters.

In 4.4.1 this works:
```
<ion-back-button defaultHref="/summary"></ion-back-button>
```
And this does not:
```
<ion-back-button defaultHref="/summary?commune=family"></ion-back-button>
```

In 4.4.0 both are working fine.

**Expected behavior:**

Addition of query parameters in `defaultHref` should not break back navigation.

**Steps to reproduce:**

Create an app with 2 pages: master-details. On details page place a back button with `defaultHref="/master-page?p1=v1"`. Open master page and navigate to details page. Click back button - it works. Navigate to details page again. Reload page (F5). Click back button - does not work.
 
**Related code:**

```
insert short code snippets here
```

**Other information:**

It worked in 4.4.0.

Possibly related bug: https://github.com/ionic-team/ionic/issues/15410
@liamdebeasi  probably related to fix for https://github.com/ionic-team/ionic/issues/18094

**Ionic info:**

```
ionic info

Ionic:

   ionic (Ionic CLI)             : 4.12.0 (C:\...\AppData\Roaming\npm\node_modules\ionic)
   Ionic Framework               : @ionic/angular 4.4.1
   @angular-devkit/build-angular : 0.13.9
   @angular-devkit/schematics    : 7.3.9
   @angular/cli                  : 7.3.9
   @ionic/angular-toolkit        : 1.5.1

System:

   NodeJS : v10.15.0 (C:\Program Files\nodejs\node.exe)
   npm    : 6.9.0
   OS     : Windows 10
```

**Comment by liamdebeasi** [comment]
Thanks for the issue! I published a nightly build of Ionic `npm i @ionic/angular@dev` that should have a fix for this if you would like to try it out and provide feedback (should be `4.5.0-dev.201905222226.d430884`) . I will also be preparing a hotfix (Ionic 4.4.2).

**Comment by liamdebeasi** [comment]
Thanks for the issue!

I published a nightly build of Ionic that has this fix if you would like to try it out and provide feedback (`npm i @ionic/angular@4.5.0-dev.201905222256.b9924c1`).

I will also be preparing a hotfix (Ionic 4.4.2).

Thanks!

**Comment by liamdebeasi** [comment]
Hey there,

I have published a 4.4.2 release with this fix.

Thanks for reporting this!

**Comment by astec** [comment]
I confirm it has been fixed in 4.4.2.

Thanks for the quick turn around.

P.S. @liamdebeasi you are the best!
